cluster/cluster_test.go

```diff
@@ -19,9 +39,103 @@ func TestJoin(t *testing.T) {
 	require.NoError(t, err)
 	defer c.Close()
 
-	t.Run("test registery contains expected services", func(t *testing.T) {
+	t.Run("test registry contains expected services", func(t *testing.T) {
 		services, err := c.Registry.Services(ctx)
 		require.NoError(t, err)
 		require.NotEmpty(t, services["ping"])
 	})
 }
+
+func (suite *ClusterSuite) TestMemberAdd() {
+	t := suite.T()
+
+	cfg, err := ConfigFromFile("./testdata/ping.yml")
+	require.NoError(t, err)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	c, err := Join(ctx, cfg)
+	require.NoError(t, err)
+	defer c.Close()
+
+	t.Run("test member list contains expected number of servers", func(t *testing.T) {
+		members, err := c.MemberList(ctx)
+		require.NoError(t, err)
+		require.Equal(t, 1, len(members))
+	})
+
+	t.Run("test member add successfully adds new member", func(t *testing.T) {
+		lcfg, err := embed.ConfigFromFile(fmt.Sprintf("./testdata/%s", cfg.EtcdConfigFile))
```

Comment:
You should be able to get `lcfg` from `cfg.etcdConfig`. Our `ConfigFromFile` method also parses the etcd config